include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -917,8 +917,8 @@ namespace VectorTools
       additional_data.mapping_update_flags = (update_values | update_JxW_values);
       MatrixFree<dim, Number> matrix_free;
       matrix_free.reinit (mapping, dof, constraints,
-                          QGauss<1>(fe_degree+1), additional_data);
-      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, 1, Number> MatrixType;
+                          QGauss<1>(fe_degree+2), additional_data);
+      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, fe_degree+2, 1, Number> MatrixType;
```

Comment:
You are right that projection is likely not a bottleneck. But isn't the current only going to the *fast* matrix-free path when the given quadrature formula to project() uses `fe_degree+2` points and through the slow matrix-based variants otherwise? My point was that we should provide the fast path for both of them.

I'm fine with keeping the tests at `fe_degree+2`. One does under-integrate on non-Cartesian meshes, but I did not observe any problems in any of my applications, so I almost never use more than `fe_degree+1` (only for nonlinear terms in fluid dynamics) and would indeed be surprised if projection did not work in that case. (Related to #3262 which I did not yet have the time to look at more closely...)